File: problems-extend-adapt/1401-1500/1490_deep-copy-an-n-ary-tree/solution_recursive.ts
function copyTree(root: Node | null): Node | null {
    if (root === null) {
        return null;
    }
    const clone = new Node(root.val);
    for (const child of root.children) {
        clone.children.push(copyTree(child));
    }
    return clone;
}

File: problems/1401-1500/1490_deep-copy-an-n-ary-tree/solution_recursive.java
import java.util.ArrayList;

class Solution {

    public Node copyTree(Node root) {
        if (root == null) {
            return null;
        }
        Node clone = new Node(root.val, new ArrayList<>());
        for (Node child : root.children) {
            clone.children.add(copyTree(child));
        }
        return clone;
    }
}
